src/main/java/thePackmaster/SpireAnniversary5Mod.java

```diff
@@ -743,35 +786,73 @@ public void update() {
     }
 
     @Override
-    public ArrayList<String> onSave() {
+    public SpireAnniversary5SaveWrapper onSave() {
+        SpireAnniversary5SaveWrapper wrapper = new SpireAnniversary5SaveWrapper();
         ArrayList<String> packIDs = new ArrayList<>();
         for (AbstractCardPack pack : currentPoolPacks) {
             packIDs.add(pack.packID);
         }
-        return packIDs;
+        wrapper.currentPoolPacks = packIDs;
+        wrapper.currentHat = Hats.currentHat;
+        return wrapper;
     }
 
     @Override
-    public void onLoad(ArrayList<String> strings) {
+    public void onLoad(SpireAnniversary5SaveWrapper saved) {
         currentPoolPacks.clear();
+        ArrayList<String> strings = saved.currentPoolPacks;
         if (strings != null) {
             for (String s : strings) {
                 currentPoolPacks.add(packsByID.get(s));
             }
         }
+
+        BaseMod.logger.info("Loading run. Hat: " + saved.currentHat);
+        if (saved.currentHat != null) {
+            Hats.currentHat = saved.currentHat;
+            Hats.addHat(true, saved.currentHat);
+        }
     }
 
     @Override
     public void receiveStartGame() {
         BaseMod.removeTopPanelItem(currentRunCardsTopPanelItem);
         if (AbstractDungeon.player.chosenClass == ThePackmaster.Enums.THE_PACKMASTER) {
             BaseMod.addTopPanelItem(currentRunCardsTopPanelItem);
+
+            Hats.atRunStart();
+            BaseMod.logger.info("completed start of game hats");
         }
+
     }
 
     public static class Enums {
         @SpireEnum
         public static AbstractGameAction.AttackEffect EVIL;
     }
 
+    private ModPanel settingsPanel;
+
+    public static boolean allPacksMode = false;
+
+    private void initializeConfig() {
+        UIStrings configStrings = CardCrawlGame.languagePack.getUIString(makeID("ConfigMenuText"));
+
+        Texture badge = TexLoader.getTexture(makeImagePath("ui/badge.png"));
+
+        settingsPanel = new ModPanel();
+        int configPos = 600;
+        //int configStep = 40;
+
+        ModLabeledToggleButton allPacksModeBtn = new ModLabeledToggleButton(configStrings.TEXT[3], 350.0f, configPos, Settings.CREAM_COLOR, FontHelper.charDescFont, allPacksMode, settingsPanel, (label) -> {
+
+        }, (button) -> {
+            allPacksMode = button.enabled;
+            saveAllPacksMode();
+        });
+
+        settingsPanel.addUIElement(allPacksModeBtn);
+
+        BaseMod.registerModBadge(badge, configStrings.TEXT[0], configStrings.TEXT[1], configStrings.TEXT[2], settingsPanel);
```

Comment:
Cool that we have a badge and config menu now.

I'd suggest a different badge icon if it's not too much trouble, something like a backpack. Right now this is using the same book icon that the current pack top panel item uses -- might just be a convenient placeholder? It's also a bit misaligned but I assume that's just because this PR has a lot of moving parts and this one hasn't settled into place yet :smile:.

![image](https://user-images.githubusercontent.com/62083413/212525138-7f156e2b-c6a4-47ec-83f2-58551924d1b0.png)

Reply:
Yep, this was a placeholder. I figured someone with actual art skills would set up some art, but I'll actually find an aligned version so it doesn't look this bad haha